apps/web/components/eventtype/EventAdvancedTab.tsx

```diff
@@ -386,20 +392,22 @@ export const EventAdvancedTab = ({ eventType, team }: Pick<EventTypeSetupInfered
                 selectedCustomInput={selectedCustomInput}
                 onSubmit={(values) => {
                   const customInput: CustomInputParsed = {
-                    id: -1,
+                    id: Math.floor(Math.random() * 10000),
```

Comment:
whats going on here? 🤣

Reply:
we can't use uuid of type string so had to generate id of type number. number of custom inputs won't be too many so there should not be any collision.

Btw this is just temporary id till user click save. real id would be assigned after user click save and custom inputs would re render with correct id

Reply:
Can we keep the number negative? That way we can GUARANTEE avoiding collision

Reply:
It would cause this bug if we use id as -1
https://photos.google.com/share/AF1QipMCbabiovDIRlru0ASfB0uerBr-aoN4XLG0_xTwAerAzAm1FFeRA4Kp-yf-Yi4bWA/photo/AF1QipP42jgzOHOeraQ45gF0MFcHCRxcEhjfl1dfyMq1?key=ZkZ0ZnZhQ2d0UXNYdE8wSHFmaWJjNTdsaWdGRmp3 .

bug is also mentioned here https://github.com/calcom/cal.com/issues/5946. the likelihood of collision is very less as after saving we would get the correct id from server. if you want i can try to further improve the id generation  Math.floor(Math.random() * 10000).

Reply:
@zomars ok added negative